commerce-initializer-util/src/main/java/com/liferay/commerce/initializer/util/CommerceAccountsImporter.java

```diff
@@ -0,0 +1,259 @@
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.util.FileUtil;
+import com.liferay.portal.kernel.util.MimeTypesUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alec Sloan
+ */
+@Component(service = CommerceAccountsImporter.class)
+public class CommerceAccountsImporter {
+
+	public List<CommerceAccount> importCommerceAccounts(
+			JSONArray jsonArray, ClassLoader classLoader,
+			Map<String, Organization> organizations, String dependenciesPath,
+			ServiceContext serviceContext)
+		throws Exception {
+
+		List<CommerceAccount> commerceAccounts = new ArrayList<>(
+			jsonArray.length());
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			CommerceAccount commerceAccount = _importCommerceAccount(
+				jsonObject, classLoader, organizations, dependenciesPath,
+				serviceContext);
+
+			commerceAccounts.add(commerceAccount);
+		}
+
+		return commerceAccounts;
+	}
+
+	protected CommerceCountry getCommerceCountryByISO(String twoLetterISOCode) {
+		DynamicQuery dynamicQuery = _commerceCountryLocalService.dynamicQuery();
+
+		Property nameProperty = PropertyFactoryUtil.forName(
+			"twoLettersISOCode");
+
+		dynamicQuery.add(nameProperty.eq(twoLetterISOCode));
+
+		List<CommerceCountry> country =
+			_commerceCountryLocalService.dynamicQuery(dynamicQuery);
+
+		return country.get(0);
+	}
+
+	protected Organization getOrganizationsByName(
+		Map<String, Organization> organizations, String organizationName) {
+
+		return organizations.get(organizationName);
+	}
+
+	private CommerceAccount _importCommerceAccount(
+			JSONObject jsonObject, ClassLoader classLoader,
+			Map<String, Organization> organizations, String dependenciesPath,
+			ServiceContext serviceContext)
+		throws IOException, PortalException {
+
+		String name = jsonObject.getString("Name");
+		String taxId = jsonObject.getString("TaxId");
+
+		// Add Commerce Account
+
+		CommerceAccount commerceAccount =
+			_commerceAccountLocalService.addCommerceAccount(
+				name, CommerceAccountConstants.DEFAULT_PARENT_ACCOUNT_ID, null,
+				taxId, CommerceAccountConstants.ACCOUNT_TYPE_BUSINESS, true,
+				null, serviceContext);
+
+		String twoLetterISOCode = jsonObject.getString("Country");
+
+		CommerceCountry commerceCountry = getCommerceCountryByISO(
+			twoLetterISOCode);
+
+		String regionCode = jsonObject.getString("Region");
+
+		long commerceRegionId = 0;
+
+		try {
+			CommerceRegion commerceRegion =
+				_commerceRegionLocalService.getCommerceRegion(
+					commerceCountry.getCommerceCountryId(), regionCode);
+
+			commerceRegionId = commerceRegion.getCommerceRegionId();
```

Comment:
@alecsloan what uppen if no regions are founded?

Reply:
Right now it throws to log and initializer continues

`com.liferay.commerce.exception.NoSuchRegionException: No CommerceRegion exists with the key {commerceCountryId=36150, code=AB}` for example when searching for Alberta region in Canada